pkg/archive/jar.go

```diff
@@ -105,3 +105,25 @@ var StableJAROrderOfAttributeValues = ZipEntryStabilizer{
 		zf.SetContent(buf.Bytes())
 	},
 }
+
+var StableGitProperties = ZipArchiveStabilizer{
+	Name: "jar-git-properties",
+	Func: func(mr *MutableZipReader) {
+		for _, mf := range mr.File {
+			// These files contain git properties set by git-commit-id-maven-plugin.
+			// They contain many unreproducible attributes as documented
+			// [here](https://github.com/git-commit-id/git-commit-id-maven-plugin/issues/825).
+
+			// Only JSON and properties file formats are available as documented
+			// [here](https://github.com/git-commit-id/git-commit-id-maven-plugin/blob/95d616fc7e16018deff3f17e2d03a4b217e55294/src/main/java/pl/project13/maven/git/GitCommitIdMojo.java#L454).
+			// By default, these file are created in ${project.build.outputDirectory} and are hence at the root of the jar.
+			// We assume that the file name is 'git' as this is the default value for the plugin.
+			// However, the plugin allows customizing the file name.
+			if strings.HasSuffix(mf.Name, "git.properties") || strings.HasSuffix(mf.Name, "git.json") {
+				// We remove these files to ensure reproducibility.
+				mr.DeleteFile(mf.Name)
```

Comment:
alternatively, we could set the content to empty. that would ensure the file is checked for presence but not equality. that would also enable us to change this to a zip file stabilizer instead of a full archive stabilizer.

that said, I'm not sure an empty file is valid json which i could imagine causing issues.

what do you think?

Reply:
> that said, I'm not sure an empty file is valid json which i could imagine causing issues.

I looked it up and it seems [it is complicated](https://stackoverflow.com/a/22361229/11751642). I tried to give `diffoscope` and empty JSON and it is unable to use its JSON parser and hence falls back to `xxd`.

```
diffoscope a.json b.json # a.json is empty

--- a.json
+++ b.json
@@ -1 +1 @@
-00000000: 0a                                       .
+00000000: 7b22 6222 3a20 317d 0a                   {"b": 1}.
```

I can try setting the content to `{}`. That should be respected by any parser.